common.sh

```diff
@@ -242,6 +242,23 @@ set_var()
 EOF
 }
 
+# Wait up to 10 minutes for cloud-init to finish
+boot_finished_check()
+{
+    retry=20
+    until [ -f /var/lib/cloud/instance/boot-finished ]; do
+        echo "Cloud-init ongoing. Waiting..."
+        retry=$((retry - 1))
+        sleep 30
+    done
+    if [ -f /var/lib/cloud/instance/boot-finished ]; then
+        echo "Cloud-init completed successfully."
+    else
+        echo "Cloud-init failed to finish."
+        exit 1
+    fi
+}
+
 # Poll for the SSH daemon to come up before proceeding.
 # The SSH poll retries with exponential backoff.
 # The initial backoff is 30s, and doubles for each retry, until 16 minutes.
```

Comment:
You are adding one more feature to this function, could you add comments here, and (maybe) get a better name for this function since it's not testing ssh only?

Reply:
Yeah I can add more comments